modules/product/tests/src/Kernel/ProductAttributesOverviewFormatterTest.php

```diff
@@ -0,0 +1,230 @@
+    $this->installEntitySchema('commerce_product_variation_type');
+    $this->installEntitySchema('commerce_product');
+    $this->installEntitySchema('commerce_product_type');
+    $this->installConfig(['commerce_product']);
+
+    $this->productDefaultDisplay = commerce_get_entity_display('commerce_product', 'default', 'view');
+    $this->attributeDefaultDisplay = commerce_get_entity_display('commerce_product_attribute_value', 'color', 'view');
+    $this->productViewBuilder = $this->container->get('entity_type.manager')->getViewBuilder('commerce_product');
+
+    $attribute = ProductAttribute::create([
+      'id' => 'color',
+      'label' => 'Color',
+    ]);
+    $attribute->save();
+    $this->container->get('commerce_product.attribute_field_manager')->createField($attribute, 'default');
+
+    EntityViewMode::create([
+      'id' => 'commerce_product.catalog',
+      'label' => 'Catalog',
+      'targetEntityType' => 'commerce_product',
+    ])->save();
+
+    $this->container->get('router.builder')->rebuildIfNeeded();
+
+    $attribute_values = [];
+    $attribute_values['cyan'] = ProductAttributeValue::create([
+      'attribute' => $attribute->id(),
+      'name' => 'Cyan',
+    ]);
+    $attribute_values['cyan']->save();
+    $attribute_values['yellow'] = ProductAttributeValue::create([
+      'attribute' => $attribute->id(),
+      'name' => 'Yellow',
+    ]);
+    $attribute_values['yellow']->save();
+
+    $this->attributeDefaultDisplay->setComponent('name', [
+      'label' => 'above',
+    ]);
+    $this->attributeDefaultDisplay->save();
+
+    $variation1 = ProductVariation::create([
+      'type' => 'default',
+      'sku' => strtolower($this->randomMachineName()),
+      'price' => [
+        'amount' => 999,
+        'currency_code' => 'USD',
+      ],
+      'attribute_color' => $attribute_values['cyan'],
+    ]);
+    $variation2 = ProductVariation::create([
+      'type' => 'default',
+      'sku' => strtolower($this->randomMachineName()),
+      'price' => [
+        'amount' => 999,
+        'currency_code' => 'USD',
+      ],
+      'attribute_color' => $attribute_values['yellow'],
+    ]);
+    /** @var \Drupal\commerce_product\Entity\ProductInterface $product */
+    $this->product = Product::create([
+      'type' => 'default',
+      'title' => 'My product',
+      'variations' => [$variation1, $variation2],
+    ]);
+    $this->product->save();
+  }
+
+  /**
+   * Test the formatters rendered display.
+   */
+  public function testFormatterDisplay() {
+    $this->productDefaultDisplay->setComponent('variations', [
+      'type' => 'commerce_product_attributes_overview',
+      'settings' => [
+        'attributes' => ['color' => 'color'],
+        'view_mode' => 'default',
+      ],
+    ]);
+    $this->productDefaultDisplay->save();
+
+    $build = $this->productViewBuilder->view($this->product);
+    $this->render($build);
+
+    $this->assertRaw('<div>
+    <div>Variations</div>
```

Comment:
The problem is there are no classes, and field injection is working. So asserting for just a color can also catch the variation titles.